### PR TITLE
Add WorkMedia to Cardigan

### PR DESCRIPTION
**Pull request by davidpmccormick**
screen shot 2018-11-21 at 12 04 38

### CODE REVIEW

File: cardigan/.storybook/webpack.config.js [hunk 1]
```diff
@@ -21,7 +21,8 @@ module.exports = (storybookBaseConfig, configType) => {
         ],
         plugins: [
           'babel-plugin-react-require',
-          '@babel/plugin-proposal-class-properties'
+          '@babel/plugin-proposal-class-properties',
+          '@babel/plugin-syntax-dynamic-import'
```
Comment (jamesgorrie) [inline cardigan/.storybook/webpack.config.js]: nice, I could't quite figure this before.

File: catalogue/webapp/components/ImageViewer/ImageViewer.js [hunk 2]
```diff
@@ -126,27 +119,13 @@ type ImageViewerState = {|
...
-  }
-
-  viewButtonMountedHandler() {
-    this.setState(prevState => ({
-      viewButtonMounted: !prevState.viewButtonMounted
-    }));
-  }
+  state = {
+    showViewer: false,
+    mountViewButton: false,
```
Comment (jamesgorrie) [inline catalogue/webapp/components/ImageViewer/ImageViewer.js]: This as a boolean state name doesn't make much sense to me?
Could we keep to `isViewButtonMounted` and (I think this should be) `shouldMountViewButton`?

Comment (jamesgorrie) [reply catalogue/webapp/components/ImageViewer/ImageViewer.js]: (or if you prefer not to as this component needs a bit of love/)

**Comment by jamesgorrie** [comment]
Saying I approve, cardigan and root tests are down.

**Comment by davidpmccormick** [comment]
Not sure how to solve the `Can't resolve 'react' ` issue… I'm assuming it has to do with the dynamic import

**Comment by jamesgorrie** [comment]
> I'm assuming it has to do with the dynamic import

If we remove the dynamic import, does it work? Doesn't quite feel like that's the issue.

File: .gitignore [hunk 3]
```diff
@@ -23,3 +23,5 @@ pa11y-report.html
 
 # Next export
 out/
+
+.DS_Store
```
Comment (jamesgorrie) [inline .gitignore]: Keeping the gitignore specific to the project keeps it clean, shouldn't these things live in our global gitignore?

(says while knowing he checked it in)

File: common/views/components/ImageViewer/ImageViewerImage.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
 // @flow
 import {Component} from 'react';
-import {convertImageUri, convertIiifUriToInfoUri} from '@weco/common/utils/convert-image-uri';
-import {spacing} from '@weco/common/utils/classnames';
+import {convertImageUri, convertIiifUriToInfoUri} from '../../../utils/convert-image-uri';
+import {spacing} from '../../../utils/classnames';
 import openseadragon from 'openseadragon';
```
Comment (jamesgorrie) [inline common/views/components/ImageViewer/ImageViewerImage.js]: We should have `openseadragon` in `common/pacakge.json` too no?

File: .gitignore (hunk 3 above)
Comment (davidpmccormick) [reply .gitignore]: haha – well it is in _my_  global .gitignore

Don't think it does any harm to be in here, though?

File: common/views/components/ImageViewer/ImageViewerImage.js (hunk 4 above)
Comment (davidpmccormick) [reply common/views/components/ImageViewer/ImageViewerImage.js]: hmmmmm… probably – will investigate

Comment (davidpmccormick) [reply common/views/components/ImageViewer/ImageViewerImage.js]: We do: https://github.com/wellcometrust/wellcomecollection.org/blob/cardigan-work-media/common/package.json#L30

**Pull request merged**